File: Jhanoo/02/[PGS - Lv2] 의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> m = new HashMap<>();
        
        for (String[] c : clothes) {
            m.put(c[1], m.getOrDefault(c[1], 0) + 1);
        }
        
        int answer = 1;
        
        for (int count : m.values()) {
            answer *= count + 1;
        }
        
        return answer - 1;
    }
}

File: hyuksanglee/02/[PGS - Lv2] 의상.js
function solution(clothes) {
    var answer = 1;
    const map ={};
    for(const [name, type] of clothes){
        map[type] = (map[type] || 0) +1;
    }
    
    
    for(const count of Object.values(map)){
        answer *= (count+1)   
    }
    return answer-1;
}
